text/0000-imply-option.md

```diff
@@ -0,0 +1,139 @@
+- Feature Name: imply-option
+- Start Date: 2017-10-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This is an RFC to reduce common boiler plate code when making use of the `Option` type. Similar in intention and motivation to the `Try` trait for `Result`.
+
+# Motivation
+[motivation]: #motivation
+
+This addition will increase the legibility of code segments and assist in defining the thought processes and motivations of programmers through code. The use cases of this addition are solutions which are expressable in the following predicate form:
+```
+    P(x) : Predicate on `x`.
+    F(y) : Function on `y`
+    P(x) -> F(y)
+```
+Or the following Rust psudocode:
+```
+    if P(x) {
+        Some(F(y))
+    } else {
+        None
+    }
+```
+The outcome of this addition will reduce repeated code which introduces bugs during refactoring and present the thought process of the programmer in a clearer fashion through their code.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The `Option` type is useful when your code may or may not yield a return value.
+Such code may looks similar to this:
+```
+    let x = 0;
+    
+    if x == 0 {
+        Some(x)
+    } else {
+        None
+    }
+```
+However only the `if` branch of this code segment is the important part we're concerned about in our code:
+```
+    if x == 0 {
+        Some(x)
+    }
+```
+But the `else` branch is required for returning `None` value if `x == 0` evaluates to false.
+Fortunately Rusts `Option` type has functionality to get rid of the unecessary code:
```

Comment:
This suggests rust already has this functionality and that at this point, the point of this proposal isn't clear at all.

Reply:
If you read the Template for RFCs provided in the repository, the `Guide-level Explanation` is meant to be an explanation by example as if it already existed in the language and it was being taught to someone who is new to Rust, similar in format to rust-by-example.